internal/datasource/os/files.go

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright (c) F5, Inc.
+ *
+ * This source code is licensed under the Apache License, Version 2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package os
+
+import (
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"os"
+	"path"
+
+	"github.com/nginx/agent/v3/api/grpc/instances"
+)
+
+func WriteFile(fileContent []byte, filePath string) error {
+	if _, err := os.Stat(filePath); os.IsNotExist(err) {
+		slog.Debug("File does not exist, creating new file", "file", filePath)
+		err = os.MkdirAll(path.Dir(filePath), 0o750)
+		if err != nil {
+			return fmt.Errorf("error creating directory, directory: %v, error: %v", path.Dir(filePath), err)
```

Comment:
Maybe lower the verbosity of the error messages slightly (leads to shorter log entries later)?
```suggestion
			return fmt.Errorf("error creating directory %s: %w", path.Dir(filePath), err)
```
Same applies to the error on line 31.